### ISSUE TITLE
Deepspeed checkpoints are completely undertrained

### ISSUE THREAD

**Issue by storuky** [kohya-ss/sd-scripts#1217]
Hey @kohya-ss and @BootsofLagrangian 
Thank you for your hard work on Deepspeed integration!  As I see this feature in dev branch.

I tried to use deepspeed in my pipelines but got some issues.
I seems that result checkpoints are undertrained with deepspeed. Can it be related to non-synchronised gradients issue? I'm currently in dev branch with the latest version.
Here is one image in dataset:
![228-image-227](https://github.com/kohya-ss/sd-scripts/assets/11155525/d5ba5c2f-258a-4c89-bc41-23f7ef6c45c6)
And here are results of same setting with and without deepspeed:
![xyz_grid-0004-3934512569](https://github.com/kohya-ss/sd-scripts/assets/11155525/b6f45bd5-8931-4ce6-8cee-c1ff5a0edeca)

The difference between these 2 versions only in 3 flags:
```
--deepspeed --zero_stage=2 --offload_optimizer_device="cpu"
```
I'm confused...

**Comment by BootsofLagrangian** [comment]
First, thanks to report issue. Is there similar phenomena on different dataset?

**Comment by storuky** [comment]
@BootsofLagrangian yes, on all datasets when using LION optimizer.
I'm not sure, maybe LION optimizer should not work as good as Adam's optimizers with Deepspeed... But it doesn't break a training process so I expect it to work...

**Comment by BootsofLagrangian** [comment]
> @BootsofLagrangian yes, on all datasets when using LION optimizer. I'm not sure, maybe LION optimizer should not work as good as Adam's optimizers with Deepspeed... But it doesn't break a training process so I expect it to work...

Yes, DeepSpeed officially supports Adam, AdamW and Fused Adam only. The behavior of other optimizer is unpredictable.

Can you try DeepSpeed with Adam or AdamW? not 8bit or bitsandbytes family.
